### ISSUE TITLE
Recategorize functions into modules and programs

### ISSUE THREAD

**Issue by darozak** [Darwin-River/Ex-Machinis#247]
I should re-categorize functions into modules and programs rather than vocabularies and dictionaries because the former terms are easier to understand in the context of current programming environments and are more consistent with the way that I'm using FORTH "words".

I'll also need to rename resource modules as units to avoid using the same word (i.e. modules) in two different contexts.

**Comment by darozak** [comment]
I recategorized vocabularies and dictionaries as modules and programs in the wiki.  To avoid the confusion of also referring to certain types of resources as modules, I reclassified systems as components and modules as systems in the resources and protocols sections of the wiki.

I still need to update the wiki tutorials so that they refer to functions, modules, and programs rather than words, vocabularies, and dictionaries.
